controllers/ocirepository_controller.go

```diff
@@ -593,9 +593,11 @@ func (r *OCIRepositoryReconciler) keychain(ctx context.Context, obj *sourcev1.OC
 		}
 	}
 
-	// if no pullsecrets available return DefaultKeyChain
+	// if no pullsecrets available return nil
+	// we don't return a DefaultKeyChain as we should not rely on the default
+	// docker config file
 	if len(pullSecretNames) == 0 {
-		return authn.DefaultKeychain, nil
+		return nil, nil
```

Comment:
I think we should return `authn.Anonymous` to not change the behaviour, this is what DefaultKeychain currently does since there is no docker config in the container.